command/workspace_new.go

```diff
@@ -75,6 +75,9 @@ func (c *WorkspaceNewCommand) Run(args []string) int {
 	}
 
 	states, err := b.States()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Failed to get configured named states: %s", err))
```

Comment:
This should `return 1` here too, which we were probably just catching at the next command that errored.